packages/@react-aria/autocomplete/src/useSearchAutocomplete.ts

```diff
@@ -66,6 +66,7 @@ export function useSearchAutocomplete<T>(props: AriaSearchAutocompleteProps<T>,
     autoComplete: 'off',
     onClear: () => {
       state.setInputValue('');
+      state.selectionManager.setSelectedKeys(new Set());
```

Comment:
Noticed that `onInputChange` is being called multiple times with different values when using the Esc key:
<img width="615" alt="image" src="https://user-images.githubusercontent.com/9661127/216679861-f8377538-fce3-4489-8c9e-a93f93937ffa.png">

This seems to be because of `useComboBox`'s Esc key handling which calls `state.revert`. Even if we clear the selected key here, that state update doesn't happen yet for `state.revert` which will cause the second onInputChange call. Ideally we'd drop the `Esc` key handling from useComboBox but keep the rest of it.

Reply:
good catch, we should get a test for this as well